onnxruntime/test/providers/qnn/matmul_test.cpp

```diff
@@ -198,6 +198,26 @@ TEST_F(QnnCPUBackendTests, DISABLED_MatMulOp) {
                     ExpectedEPNodeAssignment::All, 18);
 }
 
+TEST_F(QnnCPUBackendTests, MatMulOp_2D_x_2D_Initializer) {
+  RunMatMulOpOpTest(TestInputDef<float>({2, 3}, false, {-10.0f, -4.0f, -2.0f, 0.0f, 5.0f, 10.0f}),
+                    TestInputDef<float>({3, 2}, true, {-10.0f, -6.0f, -1.0f, 0.0f, 3.0f, 10.0f}),
+                    ExpectedEPNodeAssignment::All, 18);
+}
+
+TEST_F(QnnCPUBackendTests, DISABLED_MatMulOp_3D_x_2D) {
```

Comment:
Can you please add a comment explaining why this test needs to be disabled? Is it a bug in QNN API?

Reply:
Actually most of the UTs are disabled. ONNX's MatMul supports inputs in any rank, but QNN's MatMul supports tensors with rank >= 2 only. I've refactored the code and added more UTs. But it may fail on specific platforms. Will see and adjust the UTs.